### PR TITLE
Implement DefaultResponse commands

### LINKED ISSUE #1110: Add sending of "default response" commands as needed
#### Problem
Right now `chipZclSendDefaultResponse` in `src/app/plugin/binding-chip/chip.cpp` is a no-op.  Presumably it should send a default response.

 #### Proposed Solution
The actual sending of the response will presumably need to be handled in the relevant app code.  So I suspect we'll want to remove this no-op function and then fix any apps that stop compiling as a result (e.g. the wifi-echo server) to implement `chipZclSendDefaultResponse`.

Those implementations will need to do two things:
1. Encode the actual response into a buffer.  This should be handled by something in CHIP code, ideally.  That is, we should have an API exposed from libchip for doing this.
2. Send the resulting buffer.  This can use the existing CHIP infrastructure that the echo server already uses.

One issue implementations might run into is how to connect up the callback to `chipZclSendDefaultResponse` to the right `chipZclProcessIncoming` callsite.  Can we rely on any calls to `chipZclSendDefaultResponse` happening before `chipZclProcessIncoming` returns, so we can effectively handle this via pushing relevant state onto some sort of stack or something?  

@tecimovic @sagar-apple @bhaskar-apple

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
#### Problem
We don't actually send DefaultResponse commands right now.

 #### Summary of Changes
Hook up DefaultResponse commands to the CHIP network stack.

fixes https://github.com/project-chip/connectedhomeip/issues/1110 https://github.com/project-chip/connectedhomeip/issues/1054

### CODE REVIEW

**Comment by bzbarsky-apple** [comment]
Note: this is on top of https://github.com/project-chip/connectedhomeip/pull/1913 for now, hence draft.

File: examples/chip-tool/main.cpp
Comment (rwalker-apple) [inline examples/chip-tool/main.cpp:103]: VerifyOrExit() would work here, too instead of an early return?

Comment (rwalker-apple) [reply examples/chip-tool/main.cpp:103]: This is "ExtractDataModelMessage()" or maybe "Parse"?

File: examples/wifi-echo/server/esp32/main/DataModelHandler.cpp
Comment (rwalker-apple) [inline examples/wifi-echo/server/esp32/main/DataModelHandler.cpp:47]: this could take an Exchange (something that has to be built, might have a `SendResponse()` method on it?) instead of the raw information for where to send a response

**Review by rwalker-apple — APPROVED** [review]
cool that we have responses now

File: examples/wifi-echo/server/esp32/main/DataModelHandler.cpp
Comment (bzbarsky-apple) [reply examples/wifi-echo/server/esp32/main/DataModelHandler.cpp:47]: Right, that's the discussion that @woody-apple wants to have, I think.  ;)

File: examples/chip-tool/main.cpp
Comment (bzbarsky-apple) [reply examples/chip-tool/main.cpp:103]: That doesn't compile, because it involves a goto across the initialization of `messageLen`.  I could probably make it work with enough contortion, but I'm not sure it's worth it.

**Review by bhaskar-apple — COMMENTED** [review]
LGTM @bzbarsky-apple

**Review by woody-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@woody-apple Now that https://github.com/project-chip/connectedhomeip/pull/1913 landed, this can land too.  And once this does, I can un-draft https://github.com/project-chip/connectedhomeip/pull/1927

File: src/app/chip-zcl-zpro/command-encoder/decoder.c
Comment (andy31415) [inline src/app/chip-zcl-zpro/command-encoder/decoder.c:111]: we should add a constant like `constexpr size_t kEmberAppsFrameSize = 13;` or similar

**Review by andy31415 — APPROVED** [review]

File: src/app/chip-zcl-zpro/command-encoder/decoder.c
Comment (bzbarsky-apple) [reply src/app/chip-zcl-zpro/command-encoder/decoder.c:111]: Going to do that, but also filed https://github.com/project-chip/connectedhomeip/issues/1980 to make all this go away.

**Comment by bzbarsky-apple** [comment]
@jelderton @saurabhst @chrisdecenzo Any opinions? I need one more code owner approval to land this...

**Review by saurabhst — APPROVED** [review]

**Pull request merged**
